### DISCUSSION TITLE
Any success on React/TypeScript?

### DISCUSSION THREAD

**Discussion by lba-ben** [hiukim/mind-ar-js#292] [Q&A]
Has anyone successfully integrated mind-ar with React with TypeScript? I am attempting to do this but I'm getting stuck. I renamed the `MindARViewer.js` to `MindARViewer.tsx` to start making any necessary updates. The initial issue was with A-Frame, where TypeScript generated errors around the A-Frame component tags `<a-scene>` ,`<a-entity>`, etc. I tried installing `@types/aframe`, but it doesn't seem to include some needed references,  like `<a-camera>`. Looked around but found no immediate solution to that. I renamed the MindARView file back to `.js` and I am able to access my app, and the `sceneRef.current` object seems to be set correctly, but it has no `systems`. So `sceneEl.systems["mindar-image-system"]` returns undefined.

Has anyone successfully integrated Mind-AR with a React/TypeScript project or can anyone suggest a potential path forward?

**Comment by lsproule** [comment]
I just turned off typescript in the file, and it worked fine
